arc/database/client.py

```
@@ -658,28 +677,33 @@ async def record_outcomes_batch(self, outcomes: List[Dict[str, Any]]) -> List[st
         outcome_ids = [str(uuid.uuid4()) for _ in outcomes]
 
-        # Validate all trajectories and ensure scenarios exist
-        for outcome in outcomes:
-            if "trajectory" in outcome:
-                self._validate_trajectory(outcome["trajectory"])
+        # Process outcomes and prepare batch data
+        batch_data = []
+        for i, outcome in enumerate(outcomes):
+            # Augment and validate trajectory
+            trajectory = self._augment_trajectory(
+                outcome.get("trajectory", {}), 
```

Ruff (0.11.9)

Trailing whitespace

Remove trailing whitespace

(W291)